skimage/exposure/_adapthist.py

```diff
@@ -70,6 +70,10 @@ def equalize_adapthist(image, kernel_size=None,
     .. [1] http://tog.acm.org/resources/GraphicsGems/
     .. [2] https://en.wikipedia.org/wiki/CLAHE#CLAHE
     """
+
+    if clip_limit == 1.0:
+        return image  # is OK, immediately returns original image.
```

Comment:
```suggestion
        return img_as_float(image)  # convert to float for concistancy
```

Just to be clear of I would prefer, but we can debate :)

Reply:
@sciunto `_clahe` always returns an `uint`. The output of `equalize_adapthist`, that receives `_clahe`, [converts to `float` already](https://github.com/scikit-image/scikit-image/blob/203cf03f8c523ab00ed13c58b2fa77eb7f51ee11/skimage/exposure/_adapthist.py#L89)... this would be redundant, I guess.

Reply:
but if clip_limit==1, you bypass _clahe, if I'm correct. So we need it here as well for the case of on input with an integer dtype.

Reply:
... nevermind, @sciunto. You are right, I saw it before and I "un-saw" it somehow :sweat_smile: @rfezzani could you fix it, please? @sciunto could write in a few words what I couldn't explain before — and even find in the code at some point :)

Reply:
I already proposed that in a previous comment:
> For a coherent behavior, maybe a call to img_as_float before return is possible, but is it really what we want to do?

Reply:
> I already proposed that in a previous comment:

Yes, and I'm sorry. As I said,

> I saw it before and I "un-saw" it somehow :sweat_smile: